instrumentation/mongodb/src/main/java/brave/mongodb/MongoDBTracing.java

```diff
@@ -0,0 +1,132 @@
+    return new Builder(tracing);
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  public CommandListener commandListener() {
+    return new TraceMongoCommandListener(this);
+  }
+
+  public static final class Builder {
+    final Tracing tracing;
+    // See https://docs.mongodb.com/manual/reference/command for the command reference
+    static final List<String> DEFAULT_COMMANDS_WITH_COLLECTION_NAME = Arrays.asList("aggregate", "count", "distinct",
+      "mapReduce", "geoSearch", "delete", "find", "findAndModify", "insert", "update", "collMod", "compact",
+      "convertToCapped", "create", "createIndexes", "drop", "dropIndexes", "killCursors", "listIndexes", "reIndex");
+    final Set<String> commandsWithCollectionName = new HashSet<>();
+    int maxAbbreviatedCommandLength = 1000;
+
+    Builder(Tracing tracing) {
+      if (tracing == null) throw new NullPointerException("tracing == null");
+      this.tracing = tracing;
+      commandsWithCollectionName.addAll(DEFAULT_COMMANDS_WITH_COLLECTION_NAME);
+    }
+
+    Builder(MongoDBTracing mongoDBTracing) {
+      tracing = mongoDBTracing.tracing;
+      maxAbbreviatedCommandLength = mongoDBTracing.maxAbbreviatedCommandLength;
+      commandsWithCollectionName.addAll(mongoDBTracing.commandsWithCollectionName);
+    }
+
+    /**
+     * Sets how many characters of the MongoDB command to report in the "mongodb.command" tag. Defaults to 1000.
+     *
+     * If zero, command reporting will be disabled. The command name will still be reported as "mongodb.command.name".
+     *
+     * Set it to a large number (ex. {@link Integer#MAX_VALUE}) to disable truncation.
+     */
+    public Builder maxAbbreviatedCommandLength(int maxAbbreviatedCommandLength) {
+      if (maxAbbreviatedCommandLength < 0) throw new IllegalArgumentException("maxAbbreviatedCommandLength < 0");
+      this.maxAbbreviatedCommandLength = maxAbbreviatedCommandLength;
+      return this;
+    }
+
+    /**
+     * Clear the allow-list of command names for which tracing will attempt to extract the collection/view name from
+     * the argument.
+     */
+    public Builder clearCommandsWithCollectionName() {
+      commandsWithCollectionName.clear();
+      return this;
+    }
+
+    /**
+     * Adds a MongoDB command name to the allow-list, indicating that if the command's argument is a string, the
+     * argument contains the collection/view name. Tracing will only attempt to extract the collection/view name
+     * argument from allow-listed commands.
+     *
+     * The default allow-list is a set of commonly used commands ({@link #DEFAULT_COMMANDS_WITH_COLLECTION_NAME}) that
+     * operate on collections/views.
+     *
+     * To replace the list completely, don't forget to call {@link #clearCommandsWithCollectionName()} first.
+     *
+     * @param commandName command name to add to the allow-list
+     */
+    public Builder addCommandWithCollectionName(String commandName) {
```

Comment:
this seems pretty special case, why would a user change this? typically the impl is responsible for knowing the driver.

depending on this answer, more review to follow :)

Reply:
The idea is that in case I missed something in the default list, or if new commands get added in a newer version of MongoDB, then the user could easily add the missing commands here.

Reply:
I understand where you are going but this would be very unique as we don't do this anywhere else. We have had maintenance like this in redis before, but in general it is better to have people update the main codebase. This tension helps keep the code correct and people updating patches (vs 100 different people guessing what the missing commands were). Any first version of anything should expect to have some maintenance while things settle. Let's leave this out?

Reply:
I think I originally suggested the customization for @csabakos exact reasoning but also agree with @adriancole. So I'm fine with removing it :)